cpp/frameProcessor/include/FrameProcessorPlugin.h

```diff
@@ -66,6 +66,29 @@ class FrameProcessorPlugin : public IFrameCallback, public OdinData::IVersionedO
   /** Pointer to logger */
   LoggerPtr logger_;
 
+  /** Metadata helper function (implicitly inlined)
+   *  \param [in]  metadata - metadata struct to be read from
+   *  \param [out] message  - IpcMessage to be appended with metadata
+  */
+  void add_metadata(OdinData::IpcMessage& message, const OdinData::ParamMetadata& metadata) const
+  {
+    // TO-DO add "allowed_values" metadata
+    message.set_param("metadata/" +  metadata.path +  "/type", metadata.type);
+    message.set_param("metadata/" +  metadata.path +  "/access_mode", metadata.access_mode);
+    message.set_param("metadata/" +  metadata.path +  "/min", metadata.min);
+    message.set_param("metadata/" +  metadata.path +  "/max", metadata.max);
+    message.set_param("metadata/" +  metadata.path +  "/has_min", metadata.has_min);
+    message.set_param("metadata/" +  metadata.path +  "/has_max", metadata.has_max);
+  }
+
+  /** This is a private pure virtual method which
+   *  MUST be customized by every derived FrameProcessor 
+   *  plugin class that can append metadata.
+   *  It is essentially something in guise of a factory method.
+   * \returns a vector of ParamMetadata
+   */
+  virtual std::vector<OdinData::ParamMetadata>& get_plugin_metadata() const;
```

Comment:
```suggestion
  virtual std::vector<OdinData::ParamMetadata>& get_metadata() const;
```
plugin is redundant in this context. And as you suggested in your question, there should be two arrays and two getters, for status and config.